git_patrol.py

```diff
@@ -43,6 +43,9 @@
 GCB_ASYNC_BUILD_ID_REGEX = (
     r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}')
 
+# Limit on the total number of ref filters.
```

Comment:
Why limit the number of filters?

Reply:
tl;dr: To validate user input and preempt machine dependent failures.
Dev philosophy tangent follows...
I tend to provision limits on user specified items for validation and testing purposes. There "is" a maximum limit somewhere that's machine dependent (in this case it could be RAM, number of arguments supported by `git ls-remote`, max characters supported by bash command line, etc). However it's usually a bad user experience if they hit the machine limit before I do (in testing). Setting an arbitrary limit provides an opportunity for users to raise the issue early when they hit that limit, whence the opportunity can be used to correct their usage (if there is a better way to solve the problem), to increase the limit to handle their use case, to prompt a rewrite of the feature or all of the above. /tangent